android/src/main/java/guichaguri/trackplayer/metadata/components/MediaNotification.java

```diff
@@ -171,6 +171,11 @@ private int loadIcon(String iconName) {
 
     private Action addAction(Action instance, long mask, long action, String title, int icon,
                              List<Action> list, List<Action> compactView, boolean playing) {
+        // Add it to the compact view if it's allowed to
+        if((compactCapabilities & action) == action) {
+            compactView.add(instance);
+        }
+
```

Comment:
Why did you move it up here?

Reply:
When an action is not enabled in capabilities, they were not currently added to the compact capabilities either even if they are set in the compactCapabilities array. This is because the updateActions method return null if the capabilities is not set.

Reply:
Hum after retesting my code it seem to throw an exception when moving it here...

Reply:
ha I think that the actions in compact view should be at least the same than them in capabilities array. Sorry for that i'm gonna put the code down where it was